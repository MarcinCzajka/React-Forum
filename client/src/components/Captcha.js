import React from 'react';
import { Helmet } from "react-helmet";

class Captcha extends React.Component {
    render() {
        return (
            <div>
                <Helmet>
                    <script src="https://www.google.com/recaptcha/api.js" async defer></script>
                </Helmet>

                <div class="g-recaptcha" data-sitekey="6LdPf8kUAAAAACL0t2dMEN00RLz2NR495_4y4vzl"></div>

            </div>
        )
    }
}

export default Captcha