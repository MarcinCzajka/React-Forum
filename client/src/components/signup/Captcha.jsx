import React from 'react';
import { ReCaptcha } from 'react-recaptcha-google'

class Captcha extends React.Component {
    
    componentDidMount = () => {
        if (this.recaptcha) {
            this.recaptcha.reset();
        }
    }

    onLoadRecaptcha = () => {
        if (this.recaptcha) {
            this.recaptcha.reset();
        }
    }

    render() {
        return (
            <ReCaptcha
                ref={(el) => {this.recaptcha = el}}
                size="normal"
                data-theme="light"            
                render="explicit"
                sitekey="6LdPf8kUAAAAACL0t2dMEN00RLz2NR495_4y4vzl"
                onloadCallback={this.onLoadRecaptcha}
                verifyCallback={this.props.verifyCallback}
            />
        )
    }
}

export default Captcha