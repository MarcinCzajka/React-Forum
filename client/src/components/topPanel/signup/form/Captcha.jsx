import React from 'react';
import PropTypes from 'prop-types';
import { ReCaptcha } from 'react-recaptcha-google'

class Captcha extends React.Component {
    
    componentDidMount = () => {
        if (this.recaptcha) {
            this.recaptcha.reset();
        }
    }

    onLoadRecaptcha = () => {
        if (this.recaptcha) {
            this.recaptcha.reset();
        }
    }

    render() {
        return (
            <ReCaptcha
                ref={(el) => {this.recaptcha = el}}
                size="normal"
                data-theme="light"            
                render="explicit"
                sitekey="6LdPf8kUAAAAACL0t2dMEN00RLz2NR495_4y4vzl"
                onloadCallback={this.onLoadRecaptcha}
                verifyCallback={this.props.verifyCallback}
            />
        )
    }
}

Captcha.propTypes = {
    verifyCallback: PropTypes.func.isRequired
}

export default Captcha