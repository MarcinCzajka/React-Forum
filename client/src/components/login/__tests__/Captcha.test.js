import React from 'react';
import { create } from "react-test-renderer";
import Captcha from '../Captcha';

describe('Captcha component', () => {
    test('match snapshot', () => {
        const component = create(<Captcha />);
        expect(component).toMatchSnapshot();
    });

});