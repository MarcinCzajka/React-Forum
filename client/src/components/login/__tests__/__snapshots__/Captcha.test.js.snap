// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Captcha component match snapshot 1`] = `
<div
  data-onloadcallbackname="onloadCallback"
  data-verifycallbackname="verifyCallback"
  id="g-recaptcha"
/>
`;
